### ISSUE TITLE
Error on conditional initialization of Final class variables

### ISSUE THREAD

**Issue by ozars** [microsoft/pyright#1917]
**Describe the bug**
Conditional initialization of a Final class variable is reported as a reassignment error on elif/else branches.

**To Reproduce**
```py
from typing import Final

class A:
    x: Final[int]

    def __init__(self, a, b):
        if a:
            self.x = 1
        elif b:
            self.x = 2
        else:
            self.x = 3
```

```
No configuration file found.
No pyproject.toml file found.
stubPath /home/omer/tmp/pyright/typings is not a valid directory.
Assuming Python platform Linux
Searching for source files
Found 1 source file
/home/omer/tmp/pyright/deneme.py
  /home/omer/tmp/pyright/deneme.py:10:18 - error: "x" is declared as Final and cannot be reassigned
  /home/omer/tmp/pyright/deneme.py:12:18 - error: "x" is declared as Final and cannot be reassigned
2 errors, 0 warnings, 0 infos
Completed in 0.596sec
```

**Expected behavior**
No errors on the first correct assignment of a final variable regardless of branching.

**VS Code extension or command-line**
CLI pyright 1.1.144

**Comment by erictraut** [comment]
Thanks for the bug report. I had to go back and reread [PEP 591](https://www.python.org/dev/peps/pep-0591/). When I initially implemented the support for `Final`, I interpreted the PEP to mean that the value of a `Final` instance variable could be assigned only once within an `__init__` method. After rereading it, I think I read more into it than what was intended. I've therefore updated the logic to allow any number of assignments of a `Final` instance variable within an `__init__` method.

This change will be in the next release of pyright and pylance.

**Comment by ozars** [comment]
That's great. Thanks a lot!

**Comment by erictraut** [comment]
This is included in pyright 1.1.145, which I just published. It will also be included in the next release of pylance.
